src/input/flipper_sub.c

```diff
@@ -0,0 +1,282 @@
+	return SR_OK;
+}
+
+static int process_buffer(struct sr_input *in)
+{
+	struct context *inc;
+
+	inc = in->priv;
+
+	if (!inc->started) {
+		inc->logic_channel = sr_channel_new(in->sdi, 0, SR_CHANNEL_LOGIC, TRUE, "Raw Signal");
+		std_session_send_df_header(in->sdi);
+		sr_session_send_meta(in->sdi, SR_CONF_SAMPLERATE, g_variant_new_uint64(1000000));
+		inc->started = TRUE;
+	}
+
+	/* Process only up to the last complete to avoid processing lines where a value might be chopped/incomplete */
+	int consumed = 0;
+	gchar *last_newline = g_strrstr(in->buf->str, "\n");
+	if (last_newline) {
+		*last_newline = '\0';
+		consumed = last_newline - in->buf->str + 1;
+	}
+	/*
+	 * For each line, remove any "RAW_Data:" strings. At this point, assuming that the
+	 * data was correctly formatted, all that should be remaining would be the raw integer values.
+	 */
+	gchar **lines = g_strsplit(in->buf->str, "\n", 0);
+	int num_lines = 0;
+	if (lines)
+		num_lines = g_strv_length(lines);
+	for (int i = 0; i < num_lines; i++) {
+		gchar *occurrence = g_strstr_len(lines[i], strlen(lines[i]), RAW_DATA_STR);
+		if (occurrence) {
+			memset(occurrence, ' ', strlen(RAW_DATA_STR));
+		}
+		int ret = parse_raw_values(inc, g_strstrip(lines[i]));
+		if (ret != SR_OK) {
+			g_strfreev(lines);
+			return ret;
+		}
+	}
+	g_strfreev(lines);
+
+	/* chop off up to the point we consumed */
+	g_string_erase(in->buf, 0, consumed);
+	return SR_OK;
+}
+
+static int receive(struct sr_input *in, GString *buf)
+{
+	struct context *inc;
+
+	inc = in->priv;
+
+	sr_dbg("receive %lu bytes", buf->len);
+	g_string_append_len(in->buf, buf->str, buf->len);
+
+	if (!inc->got_header) {
+		if (!parse_header(in->buf))
+			return SR_OK;
+		sr_dbg("parsed header");
+		inc->got_header = TRUE;
+		in->sdi_ready = TRUE;
+		return SR_OK;
+	}
+
+	return process_buffer(in);
+}
+
+static int end(struct sr_input *in)
+{
+	struct context *inc;
+	int ret;
+
+	sr_dbg("end() called, frontend notified there's no more input coming");
+	inc = in->priv;
+
+	/* Finish processing any buffered data */
+	if (in->sdi_ready) {
+		ret = process_buffer(in);
+		if (ret != SR_OK) {
+			sr_dbg("process_buffer ERROR");
+			return ret;
+		}
+		sr_dbg("flushing");
+		feed_queue_logic_flush(inc->feed_logic);
+	}
+
+	/* Send DF_END when DF_HEADER was sent before */
+	if (inc->started)
+		std_session_send_df_end(in->sdi);
+
+	return SR_OK;
+}
+
+static void cleanup(struct sr_input *in)
+{
+	struct context *inc;
+
+	inc = in->priv;
+
+	feed_queue_logic_free(inc->feed_logic);
+	inc->feed_logic = NULL;
+}
+
+static int init(struct sr_input *in, GHashTable *options)
+{
+	struct context *inc;
+	(void)options; /* no options used */
+
+	inc = g_malloc0(sizeof(*inc));
+	in->sdi = g_malloc0(sizeof(*in->sdi));
+	in->priv = inc;
+	inc->feed_logic = feed_queue_logic_alloc(in->sdi, CHUNK_SIZE, 1);
+
+	return SR_OK;
+}
+
+SR_PRIV struct sr_input_module input_flipper_sub = {
+	.id = "flipper",
+	.name = "flipper",
+	.desc = "Flipper Sub-GHz v1",
```

Comment:
Flipper appears to have a bunch of file formats, and this only handles with sub-ghz v1 format. Perhaps this module should be called `flipper-subghz`?